### PR TITLE
feat: optional Obsidian layer over the memory vault (increment D)

### PR DESCRIPTION
**Pull request by howarewoo**
## What

Increment **D of 4** (final). The optional Obsidian layer over the scope-routed memory vault.

> **Stacked on #156** (distill). Base retargets as the stack lands.

## Hard reality

`obsidian eval` needs the desktop app running, so Obsidian is **never a dependency**. The grep-wikilink path (already in `recall.sh`/`doctor.sh`) stays the headless workhorse — those are unchanged and never call Obsidian.

## Changes

- `skills/woostack-init/scripts/graph.sh` (+ tests) — note links/backlinks helper. **grep by default** (headless, tested); opt-in best-effort `obsidian eval` branch (`WOOSTACK_OBSIDIAN=1` + `obsidian` CLI) that falls back to grep on any failure, never fatal. Standalone — not wired into recall's hot path.
- `skills/woostack-init/templates/obsidian/{app,graph}.json` — minimal stock vault config (vault rooted at `.woostack/`).
- `skills/woostack-init/templates/gitignore` — ignore `.obsidian/workspace*` + `cache` (per-user UI state); the shared config is tracked.
- `skills/woostack-init/SKILL.md` — `--obsidian` / `--no-obsidian` flags (with neither: prompt, default no); never-clobber scaffold step.
- `skills/woostack-init/references/memory.md` — §9 *Obsidian (optional)*; graph.sh in the scripts table.
- `AGENTS.md` — templates note.

## Verification

- 65 bash assertions green (14 new for graph: links, backlinks, .md-suffix, no-link→empty, self-exclusion, dangling target, missing-note→exit 1, **dot-in-name regex escaping**, default-uses-grep).
- Two bugs caught by independent review + fixed: unreadable-file masking (`-f`→`-r`), unescaped backlink regex.
- JSON templates valid; other four skills untouched; recall/doctor unchanged.

---

## The full stack (A0 → D)

This completes the project-memory-vault initiative:

| PR | Increment |
|---|---|
| #153 | A0 — markdown specs |
| #154 | A — woostack-init: workspace scaffold + memory contract + scripts |
| #155 | B — recall.sh + review scope-routing |
| #156 | C — distill + bootstrap/address wiring |
| #157 | D — optional Obsidian layer |

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by howarewoo** [comment]
* **#157** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/157?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/howarewoo/woostack/157?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#156** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#155** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/155?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#154** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/154?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#153** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/153?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by howarewoo — COMMENTED** [review]
## 🤖 woostack-review — CHANGES REQUESTED

Reviewed increment D (optional Obsidian layer). graph.sh (regex escaping, readability guard, never-fatal obsidian branch, bash-3.2) came back **clean** on bugs; JSON templates valid; section numbering + cross-links consistent.

**2 findings** (1 blocking doc, 1 test-coverage).

> ⚠️ Defender-only validation (prosecutor pass skipped for this batch run).

> ℹ️ Event downgraded to COMMENT (self-authored PR); CHANGES REQUESTED is the real verdict.

File: skills/woostack-init/references/memory.md [hunk 1]
```diff
@@ -156,12 +156,47 @@ The scripts live under `skills/woostack-init/scripts/` relative to the woostack
 | `scope-match.sh` | `printf '%s\n' <paths> \| bash scope-match.sh '<glob-spec>'` — prints matching paths from stdin; exits 0 if any matched, 1 if none. |
 | `build-index.sh` | `bash build-index.sh [<memdir>]` — regenerates `<memdir>/MEMORY.md` from note frontmatter; defaults to `.woostack/memory`. |
 | `doctor.sh` | `bash doctor.sh [<memdir>]` — lints the memory directory; warnings exit 0, errors exit 1. |
+| `graph.sh` | `bash graph.sh <memdir> <note> [--links\|--backlinks]` — lists a note's outbound wikilinks (`--links`, default) or the notes that link to it (`--backlinks`). Grep-based by default; see §9 for the opt-in Obsidian path. |
 
 All three scripts depend on `lib.sh` (frontmatter helpers `field()`, `note_body()`, `first_body_line()`) in the same directory. `doctor.sh` also calls `scope-match.sh`. Sourcing order: `lib.sh` first, then `scope-match.sh` if needed.
```
Comment (howarewoo) [inline skills/woostack-init/references/memory.md]: **Stale script count after adding graph.sh to §8 table**

The prose immediately after the §8 Scripts table says "All three scripts depend on `lib.sh`" but this PR adds `graph.sh` as a fourth row to that same table. A reader following the sentence after a four-row table will be misled about how many scripts use `lib.sh`, and `graph.sh` in fact does not source `lib.sh` at all — so the statement is now both numerically wrong and inaccurate about graph.sh's dependencies.

**Fix:** Change "All three scripts" to "The three scripts above (`scope-match.sh`, `build-index.sh`, `doctor.sh`)" (or equivalent phrasing that excludes `graph.sh`) so it is clear the lib.sh dependency statement does not apply to the newly-added `graph.sh`.

```suggestion
The three scripts above (`scope-match.sh`, `build-index.sh`, `doctor.sh`) depend on `lib.sh` (frontmatter helpers `field()`, `note_body()`, `first_body_line()`) in the same directory. `doctor.sh` also calls `scope-match.sh`. Sourcing order: `lib.sh` first, then `scope-match.sh` if needed. (`graph.sh` is standalone and does not source `lib.sh`.)
```

File: skills/woostack-init/scripts/tests/test-graph.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+out="$(bash "$G" "$md" a --links)"
+assert_contains "$out" "b" "default uses grep path"
+
+# dot in note name must not wildcard-match via unescaped ERE
+mk_note "$md" 'a.b.md'   $'name: a.b\ntype: pattern'   'body'
+mk_note "$md" 'decoy.md' $'name: decoy\ntype: pattern' 'links [[aXb]]'
+out="$(bash "$G" "$md" 'a.b' --backlinks)"
+assert_not_contains "$out" "decoy" "dot in note name is escaped, no wildcard false-match"
+
+rm -rf "$md"
```
Comment (howarewoo) [inline skills/woostack-init/scripts/tests/test-graph.sh:58]: **unknown-mode exit 2 branch is untested**

graph.sh documents and implements an `unknown mode` error path that prints a diagnostic and exits 2 when the third argument is neither `--links` nor `--backlinks` (graph.sh line 46). The test suite covers all other documented behaviors but never exercises this branch, so a regression that silently drops the exit code or swallows the error message would go undetected.

**Fix:** Add a test case immediately before `rm -rf "$md"`: set +e; bash "$G" "$md" a --bad-mode >/dev/null 2>&1; code=$?; set -e and assert_exit 2 "$code" "unknown mode exits 2".

File: skills/woostack-init/references/memory.md (hunk 1 above)
Comment (howarewoo) [reply skills/woostack-init/references/memory.md]: Fixed in abcca62. The §8 dependency sentence already named only build-index.sh and doctor.sh as lib.sh consumers; extended it to state scope-match.sh and graph.sh are both self-contained, so the fourth table row (graph.sh) is unambiguous.

File: skills/woostack-init/scripts/tests/test-graph.sh (hunk 2 above)
Comment (howarewoo) [reply skills/woostack-init/scripts/tests/test-graph.sh:58]: Fixed in abcca62. Added a test asserting `graph.sh <memdir> a --bad-mode` exits 2, covering the unknown-mode branch. Suite now 15/15.
